File: javaproject/app/src/main/java/com/marcosvbras/robomarket/interfaces/BaseActivityCallback.java
package com.marcosvbras.robomarket.interfaces;

import android.content.Intent;
import android.os.Bundle;

import com.marcosvbras.robomarket.DialogActions;

public interface BaseActivityCallback {

    void showDialogMessage(String message);

    void showDialogMessage(int message);

    void openActivity(Class<?> activity, boolean finishCurrentActivity);

    void openActivity(Class<?> activity, Bundle bundle, boolean finishCurrentActivity);

    void openActivityForResult(Class<?> activity, Bundle bundle, int requestCode);

    void setToolbar(int viewId, boolean displayHomeAsUpEnabled);

    void finishCurrentActivity();

    void setActivityResult(int resultCode);

    void setActivityResult(int resultCode, Intent intent);

    void setActivityResult(int resultCode, Bundle bundle);

    void showCustomAlertDialog(Object object, DialogActions actions);

}

File: javaproject/app/src/main/java/com/marcosvbras/robomarket/interfaces/BaseActivityCallback.kt
package com.marcosvbras.robomarket.interfaces

import android.content.Intent
import android.os.Bundle
import android.support.v4.app.DialogFragment

interface BaseActivityCallback {

    fun showDialogMessage(message: String)

    fun showDialogMessage(message: Int)

    fun openActivity(activity: Class<*>, finishCurrentActivity: Boolean)

    fun openActivity(activity: Class<*>, bundle: Bundle?, finishCurrentActivity: Boolean)

    fun openActivityForResult(activity: Class<*>, bundle: Bundle?, requestCode: Int)

    fun setToolbar(viewId: Int, displayHomeAsUpEnabled: Boolean)

    fun finishCurrentActivity()

    fun setActivityResult(resultCode: Int)

    fun setActivityResult(resultCode: Int, intent: Intent)

    fun setActivityResult(resultCode: Int, bundle: Bundle)

    fun showCustomAlertDialog(dialogFragment: DialogFragment)

}